### ISSUE TITLE
ValueError: operands could not be broadcast together with shapes (7931,3910) (7931,7821)

### ISSUE THREAD

**Issue by tcztzy** [multiply-org/atmospheric_correction#13]
I used to work on atmospheric correction of Landsat 8 via this package. But I struggle in the problem like following.
```python
from SIAC import SIAC_L8
import os

BASE_DIR = r"E:\Remote Sensing"

kwargs = dict(
    l8_dir=os.path.join(BASE_DIR, "Landsat", "LC08_L1TP_147031_20170515_20170525_01_T1", ""),
    mcd43=os.path.join(BASE_DIR, "MCD43", ""),
    vrt_dir=os.path.join(BASE_DIR, "MCD43", ""),
    global_dem=os.path.join(BASE_DIR, "eles", "global_dem.vrt"),
    cams_dir=os.path.join(BASE_DIR, "cams", "")
)

if __name__ == "__main__":
    SIAC_L8(**kwargs)
```

<pre>
2020-02-12 08:17:27,591 - SIAC-2.3.0 - INFO - Starting atmospheric corretion for E:\Remote Sensing\Landsat\LC08_L1TP_147031_20170515_20170525_01_T1
...
...
...
2020-02-12 09:47:18,612 - SIAC-2.3.0 - INFO - Doing corrections.
Traceback (most recent call last):
  File "C:\Users\tcztzy\Documents\GitHub\multiply-org\atmospheric_correction\main.py", line 15, in <module>
    SIAC_L8(**kwargs)
  File "<a href="https://github.com/multiply-org/atmospheric_correction/blob/master/SIAC/SIAC_L8.py#L44">C:\Users\tcztzy\Documents\GitHub\multiply-org\atmospheric_correction\SIAC\SIAC_L8.py</a>", line 44, in SIAC_L8
    aero_atmo = do_correction(*ret)
  File "<a href="https://github.com/multiply-org/atmospheric_correction/blob/master/SIAC/SIAC_L8.py#L122">C:\Users\tcztzy\Documents\GitHub\multiply-org\atmospheric_correction\SIAC\SIAC_L8.py</a>", line 122, in do_correction
    atmo._doing_correction()
  File "<a href="https://github.com/multiply-org/atmospheric_correction/blob/master/SIAC/the_correction.py#L590">C:\Users\tcztzy\Documents\GitHub\multiply-org\atmospheric_correction\SIAC\the_correction.py</a>", line 590, in _doing_correction
    ret = self._get_boa()
  File "<a href="https://github.com/multiply-org/atmospheric_correction/blob/master/SIAC/the_correction.py#L431">C:\Users\tcztzy\Documents\GitHub\multiply-org\atmospheric_correction\SIAC\the_correction.py</a>", line 431, in _get_boa
    re = list(map(self._do_chunk, bands_to_do))
  File "<a href="https://github.com/multiply-org/atmospheric_correction/blob/master/SIAC/the_correction.py#L464">C:\Users\tcztzy\Documents\GitHub\multiply-org\atmospheric_correction\SIAC\the_correction.py</a>", line 464, in _do_chunk
    toa = toa * self.ref_scale + self.ref_off
ValueError: operands could not be broadcast together with shapes (7931,3910) (7931,7821)
</pre>

**Comment by MarcYin** [comment]
I have tested with the same Landsat 8 tile with the new commit 12bf54e8bad33dbd5df27c9e2d511faf16ffeee6 and it works fine for me on a linux platform, so I assume this is likely to be solved, but please have a test on your computer again.

**Comment by tcztzy** [comment]
I found the possible cause of this issue.
All the `*VAZ_VZZ_B0X.TIF` file looks like the following:
![image](https://user-images.githubusercontent.com/12420478/75502109-d2257900-5a0c-11ea-96f4-582ff1af91be.png)
I am trying to figure out how to solve it.

**Comment by MarcYin** [comment]
If you have problem with the angles files, you could just delete them and the code will automatically regenerate them for you.

**Comment by tcztzy** [comment]
Since the workflow takes a lot of time, I am still debugging this problem. It is not related to the angles files. The problem occurred when scaling and adding offset, the shapes of np.array should be the same since they all read from TIFF file. I try to add checkpoint to this project so that I can resume from latest success step.

**Comment by MarcYin** [comment]
This problem has been solved in the newest release 131237f1a9778996406bd6d554aac4e98aeaa7d3 and it only happens in specific situation or file. It has been cause by the `goal.Warp` function, and there is a difference (sometimes) in the output array shape depending on whether you specify the spatial resolution (`xRes`, `yRes`) of the image.
